### ISSUE TITLE
Missing error message on wrong integration naming

### ISSUE THREAD

**Issue by lhein** [KaotoIO/kaoto-ui#769]
**Describe the bug**
When entering an invalid name into the integration name field there is an error marker display but no help on what is wrong exactly.

**To Reproduce**
1. Edit the integration name and put in `beerOclock`
2. try to find the error message

**Expected behavior**
I would expect to find an error message displayed as tooltip by hovering the error marker or maybe even as some text  on top or at the bottom of the text field.

**Screenshots**
![grafik](https://user-images.githubusercontent.com/435853/194027065-d3a1ba8d-bbbb-4ec9-b5a4-3f0d5ccb9d76.png)

**Comment by lordrip** [comment]
Hi, I've checked this topic and I see that there's a validation in place using a RegExp, at this moment we could only state that the RegExp failed but not get into specifics 😞, showing the RegExp to the user whenever there's a validation error doesn't seem like a good approach either 😅.

I would like to propose to split this check into multiple steps or individual functions:
1. The name only has numbers, letters, and `-`
2. All the letters are lowercase
3. The name doesn't include any special characters
4. Any other open point that the RegExp may have but I'm missing at the moment of writing

Then, if any (or all) RegExp fails, we could show a descriptive error message or a tooltip upon hovering over the exclamation mark.

For the latter, I'm not 100% sure what would be more aligned with the existing UI but if I need to vote for an approach, I would vote for having a dedicated error message below the `Input` field without needing to hover over the exclamation mark, accessibility-wise, this seems like a better approach 😃 

Lastly, I'm open to suggestions/feedback about this, or is there another solution in the works?

**Comment by lordrip** [comment]
Sure thing @kahboom , I'll check the docs and then I'll try to do a quick draft and share it with the team

**Comment by lordrip** [comment]
Hi again, I've checked the PatternFly docs and it seems that is not possible to have error messages that span multiple lines 😞, so I've explored a few options

### Option 1 - Use a single-line error message
We could show a concise error message for a validation error like in the following
![image](https://user-images.githubusercontent.com/16512618/203163463-b37aba88-3b3f-48f7-aeb7-62eda505da26.png)

There are a few things that are happening with this though:
1. The Toolbar height increases to accommodate the error message
2. The width of the input field increases to accommodate the length of the message
https://user-images.githubusercontent.com/16512618/203162739-fea98152-8e1e-496d-88c4-92f422152580.mp4

### Option 2 - Individual error messages one by one
That being said, another approach that I've thought is to set error messages one by one with the tradeoff of the user making multiple modifications until the name is finally correct. As an example:
1. The user writes This_Is_An_Invalid_Name-
3. We notify the user that `_` is not allowed in the name, the user fixes it
4. We notify the user that uppercase letters aren't allowed either
5. Lastly we notify the user that a trailing `-` is not allowed 

At this point, is very likely that we'll get an angry user, so I don't believe that is the way to go. 🙅 

### Option 3 - An error banner
There's also the possibility to include an error banner to the FormGroup, with the caveat of increasing the width of the Toolbar considerably and it doesn't look very well
![image](https://user-images.githubusercontent.com/16512618/203162161-ff55b651-3b89-495f-ba6a-93372faf88b2.png)

### Option 4 - Custom error message
Another alternative that I've explored is to implement a custom banner below the input field that overflows from the Toolbar avoiding increasing the height, the tradeoff of this approach is that it doesn't seem like an existing PatternFly pattern.
![image](https://user-images.githubusercontent.com/16512618/203161146-e2299a5d-c385-4c37-be30-11316ae23318.png)

It seems that the most compliant way to notify the user is to use **Option #1**, however, we might tackle merging the Titlebar with the Toolbar first (#937) to get more room for a potential error message.

I'll stay tuned to know what you consider is the best approach 😃

**Comment by kahboom** [comment]
@lordrip - First off, so sorry for the major delay in response. Thank you for putting together this proposal and including the examples.

I agree with you that option 1 seems to be the best option in terms of usability. I don't think we need to wait for the toolbar issue to be completed first, but we certainly can if it looks like there will be more work or changes to be made afterward. Option 4 isn't too bad either, but my concern is it not staying in line with PatternFly over time. Option 1 definitely gets my vote. 👍

**Comment by lordrip** [comment]
No, we're gonna implement it taking this issue into consideration.
